hamilton/function_modifiers/macros.py

```diff
@@ -620,6 +620,8 @@ def step(
     they will be converted to a value (a literal)
     :return: an applicable with the function applied
     """
+    # This function will be excluded from the DAG as a node since we are inserting it manually
+    fn = ignore()(fn)
```

Comment:
So, thinking about this, I'm not sure we should be injecting this with `step` -- to me this might be a parameter we want to `pipe_output`. 

The problem is we won't know whether a function is included until later (when it's referred), which feels harder to parse. So maybe remove this, or put this behind a default-off toggle?